fast_neural_style/neural_style/neural_style.py

```diff
@@ -0,0 +1,231 @@
+import argparse
+import os
+import sys
+import time
+
+import numpy as np
+import torch
+from torch.autograd import Variable
+from torch.optim import Adam
+from torch.utils.data import DataLoader
+from torchvision import datasets
+from torchvision import transforms
+
+import utils
+from transformer_net import TransformerNet
+from vgg import Vgg16
+
+
+def train(args):
+    np.random.seed(args.seed)
+    torch.manual_seed(args.seed)
+
+    if args.cuda:
+        torch.cuda.manual_seed(args.seed)
+
+    transform = transforms.Compose([
+        transforms.Scale(args.image_size),
+        transforms.CenterCrop(args.image_size),
+        transforms.ToTensor(),
+        transforms.Lambda(lambda x: x.mul(255))
+    ])
+    train_dataset = datasets.ImageFolder(args.dataset, transform)
+    train_loader = DataLoader(train_dataset, batch_size=args.batch_size)
+
+    transformer = TransformerNet()
+    optimizer = Adam(transformer.parameters(), args.lr)
+    mse_loss = torch.nn.MSELoss()
+
+    vgg = Vgg16()
+    style_transform = transforms.Compose([
+        transforms.ToTensor(),
+        transforms.Lambda(lambda x: x.mul(255))
+    ])
+    style = utils.load_image(args.style_image, size=args.style_size)
+    style = style_transform(style)
+    style = style.repeat(args.batch_size, 1, 1, 1)
+
+    if args.cuda:
+        transformer.cuda()
+        vgg.cuda()
+        style = style.cuda()
+
+    style_v = Variable(style, volatile=True)
+    style_v = utils.normalize_batch(style_v)
+    features_style = vgg(style_v)
+    gram_style = [utils.gram_matrix(y) for y in features_style]
+
+    for e in range(args.epochs):
+        transformer.train()
+        agg_content_loss = 0.
+        agg_style_loss = 0.
+        count = 0
+        for batch_id, (x, _) in enumerate(train_loader):
+            n_batch = len(x)
+            count += n_batch
+            optimizer.zero_grad()
+            x = Variable(x)
+            if args.cuda:
+                x = x.cuda()
+
+            y = transformer(x)
+
+            xc = Variable(x.data.clone(), volatile=True)
+
+            y = utils.normalize_batch(y)
+            xc = utils.normalize_batch(xc)
+
+            features_y = vgg(y)
```

Comment:
The weights of VGG are not to be updated. They were not being updated earlier because the optimizer used only the parameters of generator. I have detached the parameters of VGG as you suggested, [code](https://github.com/darkstar112358/examples/blob/24bfb3c1ec16868eb336db18a83b17fd022ea57c/fast_neural_style/neural_style/vgg.py#L25-L27). I don't know of a better way of not to require gradients.